util/util.go

```diff
@@ -268,6 +268,22 @@ func EnsureOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerRef
 	return ownerReferences
 }
 
+// ReconcileOwnerRef re-parents an object from one OnwerReference to another
+func ReconcileOwnerRef(ownerReferences []metav1.OwnerReference, target metav1.OwnerReference, source metav1.Object) []metav1.OwnerReference {
+	fi := -1
+	for index, r := range ownerReferences {
```

Comment:
Couldn't we use `indexOwnerRef` here and replace at index?

Reply:
`indexOwnerRef` has different semantics, I wanted UID matching to avoid stealing ownership of an object that *should* be deleted.

Reply:
@sethp-nr Reached any conclusions on this one?

Reply:
Oh, sorry, I missed Vince's comment – uniqueness is not guaranteed on object namespace + name across time. If a user deletes a KCP without orphaning dependents and then re-creates it, the intent I think we should honor is to delete the old machines and create a new control plane.